### ISSUE TITLE
UnityNativeGallery plugin takes some time while taking screen shot at first time

### ISSUE THREAD

**Issue by ArigarasuthanRepo** [yasirkula/UnityNativeGallery#34]
Iam using UnityNativeGallery plugin to take screenshot i n unity and android.When i  newly install the app and take a screenshot it was hanged 5 seconds and worked.But in the second time its worked.Taking screenshot at the very first time using UnityNativeGallery plugin this was arised.How can i solve it?

**Comment by yasirkula** [comment]
I didn't notice a lag while testing it myself. I'm unsure about the cause of this issue. Will keep you updated if I find something new.

**Comment by ArigarasuthanRepo** [comment]
Ok Thank you

**Comment by ArigarasuthanRepo** [comment]
Any solution for this issue?

**Comment by yasirkula** [comment]
If you just take a screenshot but don't call any NativeGallery functions afterwards, does it still freeze for 5 seconds initially?

**Comment by ArigarasuthanRepo** [comment]
How to take screenshot without NativeGallery functions?

**Comment by yasirkula** [comment]
Call the following function like this: `StartCoroutine( TakeScreenshotDontSave() );`

```csharp
private IEnumerator TakeScreenshotDontSave()
{
	yield return new WaitForEndOfFrame();

	Texture2D ss = new Texture2D( Screen.width, Screen.height, TextureFormat.RGB24, false );
	ss.ReadPixels( new Rect( 0, 0, Screen.width, Screen.height ), 0, 0 );
	ss.Apply();

	// To avoid memory leaks
	Destroy( ss );
}
```

**Comment by ArigarasuthanRepo** [comment]
Still iam not tried..I will try it and tell it.

**Comment by ArigarasuthanRepo** [comment]
Iam using button click to taking screenshot.When i click the button after new installation of my app and click a button to take screenshot it was asking permission using dialog.Is it anyway to get permission instead of while taking screenshot operation was going on.

**Comment by yasirkula** [comment]
Of course! You can use the `NativeGallery.RequestPermission()` function.

**Comment by TobiasWehrum** [comment]
Is the app supposed to pause at all when calling NativeGallery.SaveImageToGallery/NativeGallery.SaveVideoToGallery? I think I might be getting the same freeze.
